python-package/lightgbm/dask.py

```diff
@@ -1006,16 +1400,34 @@ def fit(
         sample_weight: Optional[_DaskCollection] = None,
         init_score: Optional[_DaskCollection] = None,
         group: Optional[_DaskCollection] = None,
+        eval_set: Optional[List[Tuple[_DaskMatrixLike, _DaskCollection]]] = None,
+        eval_names: Optional[List[str]] = None,
+        eval_sample_weight: Optional[List[_DaskCollection]] = None,
+        eval_init_score: Optional[List[_DaskCollection]] = None,
+        eval_group: Optional[List[_DaskCollection]] = None,
+        eval_metric: Optional[Union[Callable, str, List[Union[Callable, str]]]] = None,
+        eval_at: Optional[Iterable[int]] = (1, 2, 3, 4, 5),
```

Comment:
Sorry, I'm not familiar with typing, just saw this @jameslamb's comment.
> The `Optional` type in `typing` is not intended to document whether or not something has a default argument. `Optional[some_type]` is just a convenience for saying `Union[some_type, None]`.
   https://github.com/microsoft/LightGBM/pull/4367#discussion_r650588094

```suggestion
        eval_at: Iterable[int] = (1, 2, 3, 4, 5),
```

Reply:
Ah good call, thanks

Reply:
Addressed in 6da34f6